model/config.go

```diff
@@ -180,11 +180,11 @@ func (i *IstioRegistry) RouteRulesBySource(namespace string, instances []*Servic
 	rules := make([]*routeRuleConfig, 0)
 	for key, rule := range i.RouteRules(namespace) {
 		// validate that rule match predicate applies to source service instances
-		if rule.Match != nil {
+		if rule.Match != nil && rule.Match.Source != "" {
 			found := false
```

Comment:
@frankbu can you add a comment on top of this if block to explain the rationale? Ingress routes won't have a source field right?